src/local-state/__tests__/LocalState/rootValue.test.ts

```diff
@@ -0,0 +1,144 @@
+import { equal } from "@wry/equality";
+
+import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";
+import { LocalState } from "@apollo/client/local-state";
+
+import { gql } from "./testUtils.js";
+
+test("passes parent value as empty object to root resolver for client-only query", async () => {
+  const document = gql`
+    query Test {
+      foo @client {
+        bar
+      }
+    }
+  `;
+
+  const client = new ApolloClient({
+    cache: new InMemoryCache(),
+    link: ApolloLink.empty(),
+  });
+
+  const localState = new LocalState({
+    resolvers: {
+      Query: {
+        foo: (rootValue) => ({
+          __typename: "Foo",
+          bar: equal(rootValue, {}),
+        }),
+      },
+    },
+  });
+
+  await expect(
+    localState.execute({
+      document,
+      client,
+      context: {},
+      variables: {},
+      remoteResult: undefined,
+    })
+  ).resolves.toStrictEqualTyped({
+    data: { foo: { __typename: "Foo", bar: true } },
+  });
+});
+
```

Comment:
I believe we're currently missing an error along these lines? (This is currently failing.)

```suggestion
test("remote error, `data` is `null` - does not execute resolvers and adds an error", async () => {
  const document = gql`
    query Test {
      foo @client {
        bar
      }
    }
  `;

  const client = new ApolloClient({
    cache: new InMemoryCache(),
    link: ApolloLink.empty(),
  });

  const localState = new LocalState({
    resolvers: {
      Query: {
        foo: () => ({
          __typename: "Foo",
          bar: true,
        }),
      },
    },
  });

  await expect(
    localState.execute({
      document,
      client,
      context: {},
      variables: {},
      remoteResult: {
        errors: [{ message: "Error from server" }],
      },
    })
  ).resolves.toStrictEqualTyped({
    data: null,
    errors: [
      { message: "Error from server" },
      {
        message:
          "Could not merge data from 'foo' resolver with remote data since data was `null`.",
      },
    ],
  });
});
```

Reply:
See https://github.com/apollographql/apollo-client/pull/12614#discussion_r2095898108. This was accidentally left in.